remoulade/message.py

```diff
@@ -116,6 +119,17 @@ def cancel(self):
         backend = broker.get_cancel_backend()
         backend.cancel([self.message_id])
 
+    def set_progress(self, progress):
+        """ Set the progress of the message.
+            progress(float) number between 0 and 1 inclusive
+        """
+        if progress < 0.0 or progress > 1.0:
+            raise InvalidProgress("Progress {} is not between 0 and 1.".format(progress))
+
+        broker = get_broker()
+        backend = broker.get_state_backend()
+        backend.set_state(State(self.message_id, StateNamesEnum.Started, progress=progress))
```

Comment:
I would prefer you do not set the state name here. We are pretty confident it's started but anyway.